torchrecipes/paved_path/charnn/export.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import argparse
+
+import fsspec
+import torch
+
+parser = argparse.ArgumentParser(description="Quantize the model from a snapshot")
+parser.add_argument(
+    "-i",
+    "--input_path",
+    type=str,
+    required=True,
+    help="Snapshot path to load. It can be a local path, S3 or Google Cloud Storage URL",
+)
+parser.add_argument(
+    "-o",
+    "--output_path",
+    type=str,
+    required=True,
+    help="Snapshot path to save. It can be a local path, S3 or Google Cloud Storage URL",
+)
+parser.add_argument("-q", "--quantize", help="quantize the model", action="store_true")
+parser.add_argument(
+    "-t", "--torchscript", help="torchscript the model", action="store_true"
+)
+
+
+def main() -> None:
+    args = parser.parse_args()
+
+    fs, intput_path = fsspec.core.url_to_fs(args.input_path)
+    with fs.open(intput_path, "rb") as f:
+        model = torch.load(f, map_location="cpu")
```

Comment:
shall we use snapshot to save/load trained model as well?

Reply:
I tried but feel it causes more trouble than benefits. Snapshot save/load state_dicts from training states including model, optimizer, progress, etc. 1) we only need model here. It's a waste to pass the larger snapshot file in S3 2) to load the state_dict, we have to initialize the GPT model object here with the same config as training, which make this script less generic.

Reply:
For "Snapshot save/load state_dicts from training states including model, optimizer, progress, etc.", we should be able to create a separate snapshot that only save model params? but anyway, it sounds reasonable to avoid the overhead. maybe lets check with @yifuwang about the appropriate use case of snapshot